src/resonate/scheduler.py

```diff
@@ -558,67 +739,53 @@ def _run(self) -> None:  # noqa: C901, PLR0912
 
             assert not self._runnable_coros, "Runnables should have been all exhausted"
 
-    def _add_coro_to_awaitables(
-        self, p: Promise[Any], coro_and_promise: CoroAndPromise[Any]
-    ) -> None:
+    def _add_coro_to_awaitables(self, p: Promise[Any], coro: ResonateCoro[Any]) -> None:
         assert (
             not p.done()
         ), "If the promise is resolved already it makes no sense to block coroutine"
-        self._awaitables.setdefault(p, []).append(coro_and_promise)
-        coro_and_promise.route_info.promise.increase_blocked_on()
+        self._awaitables.setdefault(p, []).append(coro)
+
+        if isinstance(p.action, LFI) and isinstance(p.action.exec_unit, Command):
+            assert not isinstance(
+                p.action.exec_unit, CreateDurablePromiseReq
+            ), "This command is reserved for rfi."
+            self._send_pending_commands_to_processor(type(p.action.exec_unit))
+
+        coro.route_info.promise.increase_blocked_on()
+
         self._tracing_adapter.process_event(
             ExecutionAwaited(
-                promise_id=coro_and_promise.route_info.promise.promise_id,
+                promise_id=coro.route_info.promise.promise_id,
                 tick=now(),
-                parent_promise_id=coro_and_promise.route_info.promise.parent_promise_id(),
+                parent_promise_id=coro.route_info.promise.parent_promise_id(),
             )
         )
 
     def _add_coro_to_runnables(
         self,
-        coro_and_promise: CoroAndPromise[Any],
+        coro: ResonateCoro[Any],
         value_to_yield_back: Result[Any, Exception] | None,
         *,
         was_awaited: bool,
     ) -> None:
-        self._runnable_coros.append(
-            (Runnable(coro_and_promise, value_to_yield_back), was_awaited)
+        self._runnable_coros.appendleft(
+            (Runnable(coro, value_to_yield_back), was_awaited)
         )
 
     def _unblock_coros_waiting_on_promise(self, p: Promise[Any]) -> None:
         assert p.done(), "Promise must be done to unblock waiting coroutines."
         if self._awaitables.get(p) is None:
             return
 
-        for coro_and_promise in self._awaitables.pop(p):
-            coro_and_promise.route_info.promise.decrease_blocked_on()
-            if coro_and_promise.route_info.promise.is_blocked():
+        for coro in self._awaitables.pop(p):
+            coro.route_info.promise.decrease_blocked_on()
+            if coro.route_info.promise.is_blocked():
```

Comment:
Do we still need this pattern? I suspect that with the new way of handling structured concurrency, this might not be necessary, but I am unsure.

Reply:
Removed